src/Adapter/ImageThumbnailsRegenerator.php

```diff
@@ -275,10 +269,14 @@ public function regenerateNoPictureImages(string $dir, array $type, array $langu
 
         foreach ($type as $image_type) {
             foreach ($languages as $language) {
+                // We get the "no image available" in the folder of the object
                 $file = $dir . $language->getIsoCode() . '.jpg';
+
+                // If it doesn't exist, we use a fallback one in the root of img directory
                 if (!file_exists($file)) {
-                    $file = _PS_PRODUCT_IMG_DIR_ . $defaultLang->getIsoCode() . '.jpg';
+                    $file = _PS_IMG_DIR_ . 'noimageavailable.jpg';
```

Comment:
Here you change the default behaviour, the `noimageavailable` image should only be used as a last resort, if the image with the proper language from default configuration is available then it should be used in priority

It's better to have a localized default image instead of a globally english one

Reply:
Well, the thing is, that for example, for a category, it tries to use _PS_MANU_IMG_DIR_ / fr.jpg

And if it doesn't exist, it uses _PS_PRODUCT_IMG_DIR_ / fr.jpg

Or, for product, it tries to use the same image for fallback. :-)))

Reply:
Fixed